src/main/java/io/pravega/connectors/flink/ReaderCheckpointHook.java

```diff
@@ -103,7 +100,7 @@ public void restoreCheckpoint(long checkpointId, Checkpoint checkpoint) throws E
         // checkpoint can be null when restoring from a savepoint that
         // did not include any state for that particular reader name
         if (checkpoint != null) {
-            this.readerGroup.resetReadersToCheckpoint(checkpoint);
+            this.readerGroup.resetReaderGroup(ReaderGroupConfig.builder().startFromCheckpoint(checkpoint).build());
```

Comment:
I know that `resetReadersToCheckpoint` was marked deprecated but I think we should challenge that notion, and continue to use it for now.   Notice that the 'new' approach (as shown here) discards the earlier reader group configuration.  For example, a non-default value for `automaticCheckpointIntervalMillis` would be lost.